hbenl.vscode-test-explorer, ms-python.python : missing "out" folder

https://open-vsx.org/extension/hbenl/vscode-test-explorer

Normally after build it should have a `out` folder that contains the "executables" (.js files) for that extension, that gets packaged in the `.vsix`, but that's missing from what's published on open vsx. 

Building it locally, I was able to obtain that folder with these commands:
```
$> npm install 
$> npm run build
```

I see the [`npm install` is covered](https://github.com/open-vsx/publish-extensions/blob/master/publish-extensions.js#L66). I do not see that `npm run build` is called? 

If this is unexpected to be required, maybe a solution would be to support a field in `extensions.json` where one can add extra required command(s) to run to correctly build some extensions that need a bit more of a custom procedure, before the script proceeds to that packaging/publishing phase?

__update:__
for ms-python.python I was able to update the `out` folder like so:
```
$> npm install
$> npm run package
```

I think `ms-python.python` has the same problem:
```
root INFO [hosted-plugin: 21414] PLUGIN_HOST(21414): PluginManagerExtImpl/loadPlugin(/tmp/vscode-unpacked/vscode-builtin-python.vsix/extension/dist/pythonMain)
root INFO [hosted-plugin: 21414] PLUGIN_HOST(21414): PluginManagerExtImpl/loadPlugin(/home/<user>/.<prod>/extensions/ms-python.python-2020.2.3/extension/out/client/extension)
root ERROR [hosted-plugin: 21414] { Error: Cannot find module '/home/<user>/.<prod>/extensions/ms-python.python-2020.2.3/extension/out/client/extension'

The extension exists at that path, except from `out`. Another clueL the extension is a few hundred KB in size vs 40+ MB on GH release: https://github.com/microsoft/vscode-python/releases
```

--- a/README.md
+++ b/README.md
@@ -33,15 +33,17 @@ Here are all the supported values, including optional ones:
 ```js
     {
       // Unique Open VSX extension ID in the form "<namespace>.<name>"
-      "id": "ms-vscode.cpptools",
+      "id": "rebornix.ruby",
       // (OPTIONAL) The version to publish to Open VSX (defaults to the package.json version)
-      "version": "0.26.3",
+      "version": "0.27.0",
       // Repository URL to clone and publish from
-      "repository": "https://github.com/microsoft/vscode-cpptools",
+      "repository": "https://github.com/rubyide/vscode-ruby",
       // (OPTIONAL) The Git branch, tag, or commit to check out before publishing (defaults to the repository's default branch)
-      "checkout": "0.26.3",
+      "checkout": "v0.27.0",
       // (OPTIONAL) Location of the extension's package.json in the repository (defaults to the repository's root directory)
-      "location": "Extension"
+      "location": "packages/vscode-ruby-client",
+      // (OPTIONAL) Extra commands to run just before publishing to Open VSX (i.e. after "yarn/npm install", but before "vscode:prepublish")
+      "prepublish": "npm run build"
     },
 ```
 
@@ -54,7 +56,8 @@ The [publishing process](https://github.com/open-vsx/publish-extensions/blob/e70
 1. [`git clone "repository"`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L58)
 2. _([`git checkout "checkout"`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L60) if a `"checkout"` value is specified)_
 3. [`npm install`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L66) (or `yarn install` if a `yarn.lock` file is detected in the provided `"location"`)
-4. _([`ovsx create-namespace "publisher"`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L68-L74) if it doesn't already exist)_
-5. [`ovsx publish`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L76-L82) (with `--yarn` if a `yarn.lock` file was detected earlier)
+4. _(`prepublish`)_
+5. _([`ovsx create-namespace "publisher"`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L68-L74) if it doesn't already exist)_
+6. [`ovsx publish`](https://github.com/open-vsx/publish-extensions/blob/e70fb554a5c265e53f44605dbd826270b860694b/publish-extensions.js#L76-L82) (with `--yarn` if a `yarn.lock` file was detected earlier)
 
 See all `ovsx` CLI options [here](https://github.com/eclipse/openvsx/blob/master/cli/README.md).

--- a/add-extension.js
+++ b/add-extension.js
@@ -21,7 +21,7 @@ const writeFile = util.promisify(fs.writeFile);
 
 (async () => {
   const argv = minimist(process.argv.slice(2));
-  /** @type {{ extensions: { id: string, version?: string, repository: string, checkout?: string, location?: string }[] }} */
+  /** @type {{ extensions: { id: string, version?: string, repository: string, checkout?: string, location?: string, prepublish?: string }[] }} */
   const { extensions } = JSON.parse(await readFile('./extensions.json', 'utf-8'));
   const registry = new ovsx.Registry();
 

--- a/extensions.json
+++ b/extensions.json
@@ -306,6 +306,13 @@
       "repository": "https://github.com/mitaki28/vscode-clang"
     },
     {
+      "id": "ms-python.python",
+      "version": "2020.5.1",
+      "repository": "https://github.com/Microsoft/vscode-python",
+      "checkout": "2020.5.80290",
+      "prepublish": "npm run package"
+    },
+    {
       "id": "naumovs.color-highlight",
       "version": "2.3.0",
       "repository": "https://github.com/egonyans/vscode-ext-color-highlight"

--- a/extensions.json
+++ b/extensions.json
@@ -212,7 +212,8 @@
     {
       "id": "hbenl.vscode-test-explorer",
       "version": "2.19.1",
-      "repository": "https://github.com/hbenl/vscode-test-explorer"
+      "repository": "https://github.com/hbenl/vscode-test-explorer",
+      "prepublish": "npm run build"
     },
     {
       "id": "hollowtree.vue-snippets",

--- a/publish-extensions.js
+++ b/publish-extensions.js
@@ -18,7 +18,7 @@ const exec = require('./lib/exec');
 const readFile = util.promisify(fs.readFile);
 
 (async () => {
-  /** @type {{ extensions: { id: string, version?: string, repository: string, checkout?: string, location?: string }[] }} */
+  /** @type {{ extensions: { id: string, version?: string, repository: string, checkout?: string, location?: string, prepublish?: string }[] }} */
   const { extensions } = JSON.parse(await readFile('./extensions.json', 'utf-8'));
   const registry = new ovsx.Registry();
 
@@ -63,6 +63,9 @@ const readFile = util.promisify(fs.readFile);
         fs.access(path.join('/tmp/repository', 'yarn.lock'), error => resolve(!error));
       });
       await exec(`${yarn ? 'yarn' : 'npm'} install`, { cwd: '/tmp/repository' });
+      if (extension.prepublish) {
+        await exec(extension.prepublish, { cwd: '/tmp/repository' })
+      }
 
       // Create a public Open VSX namespace if needed.
       try {

--- a/upgrade-extensions.js
+++ b/upgrade-extensions.js
@@ -23,7 +23,7 @@ const dontUpgrade = [
 ];
 
 (async () => {
-  /** @type {{ extensions: { id: string, version?: string, repository: string, checkout?: string, location?: string }[] }} */
+  /** @type {{ extensions: { id: string, version?: string, repository: string, checkout?: string, location?: string, prepublish?: string }[] }} */
   const { extensions } = JSON.parse(await readFile('./extensions.json', 'utf-8'));
   const extensionsToUpgrade = extensions.filter(e => !dontUpgrade.includes(e.id))
   const extensionsToNotUpgrade = extensions.filter(e => dontUpgrade.includes(e.id));

